test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c

```diff
@@ -570,6 +611,71 @@ void test_DNS_TreatNBNS_success_nbns_non_fixed_size_buffer( void )
     ASSERT_DNS_QUERY_HOOK_CALLED();
 }
 
+/**
+ * @brief success path, formed packet is sent over the network with
+ *        vReturnEthernetFrame
+ */
+void test_DNS_TreatNBNS_success_nbns_non_fixed_size_buffer2( void )
+{
+    uint8_t pucPayload[ 300 ];
+    size_t uxBufferLength = 300;
+    uint32_t ulIPAddress;
+
+    xBufferAllocFixedSize = pdTRUE;
+    NetworkBufferDescriptor_t pxNetworkBuffer;
+    NetworkBufferDescriptor_t pxNetworkBuffer_dup;
+
+    uint8_t buffer[ 300 ];
+
+    pxNetworkBuffer.pucEthernetBuffer = buffer;
+    pxNetworkBuffer.xDataLength = 300;
+
+    pxNetworkBuffer_dup.pucEthernetBuffer = buffer;
+    pxNetworkBuffer_dup.xDataLength = 300;
+
+    hook_return = pdTRUE;
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_FLAGS_OPCODE_QUERY ); /* usFlags */
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_TYPE_NET_BIOS );      /* usType */
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_FLAGS_OPCODE_QUERY );
+    pxUDPPayloadBuffer_to_NetworkBuffer_ExpectAnyArgsAndReturn( &pxNetworkBuffer );
+    usGenerateChecksum_ExpectAnyArgsAndReturn( 4 );
+    usGenerateProtocolChecksum_ExpectAnyArgsAndReturn( 4 );
+    vReturnEthernetFrame_Expect( &pxNetworkBuffer, pdFALSE ); /* goal */
+
+    DNS_TreatNBNS( pucPayload,
+                   uxBufferLength,
+                   1234 );
+    ASSERT_DNS_QUERY_HOOK_CALLED();
+}
+
+/**
+ * @brief success path, formed packet is sent over the network with
+ *        vReturnEthernetFrame
+ */
+void test_DNS_TreatNBNS_success_nbns_non_fixed_size_buffer3( void )
+{
+    uint8_t pucPayload[ 300 ];
+    size_t uxBufferLength = 300;
+    uint32_t ulIPAddress;
+
+    NetworkBufferDescriptor_t pxNetworkBuffer;
+    NetworkBufferDescriptor_t pxNetworkBuffer_dup;
+
+    uint8_t buffer[ 300 ];
+
+    pxNetworkBuffer_dup.pucEthernetBuffer = buffer;
+    pxNetworkBuffer_dup.xDataLength = 300;
+
+    hook_return = pdTRUE;
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_FLAGS_OPCODE_QUERY ); /* usFlags */
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_CLASS_IN );           /* usType */
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_FLAGS_OPCODE_QUERY );
+
+    DNS_TreatNBNS( pucPayload,
+                   uxBufferLength,
+                   1234 );
+}
```

Comment:
No assert after the test to verify that it did what it was expected?

Reply:
nothing to assert on with this function, as it doesn't change arguments, not returns anything, it only calls other functions